src/init.d/image_checker.py

```diff
@@ -95,6 +96,12 @@ def __init__(self, job_config, secret):
             auth = image_info["auth"]
             self._init_auth_info(auth, secret)
 
+    def _get_registry_from_image_uri(self, image_uri) -> str:
+        if self._is_image_use_default_domain():
+            return DEFAULT_REGISTRY
+        index = self._image_uri.find("/")
+        return _get_registry_uri(image_uri[:index])
```

Comment:
what if the registry uses a non-root path

Reply:
If use non-root path, it will be treated use default domain

Reply:
I mean the registry "domain:5000/path/"

Reply:
Refer to https://github.com/docker/distribution/blob/742aab907b54a367e1ac7033fb9fe73b0e7344f5/reference/normalize.go#L91, seems docker only treats domain:5000 as registry path